### ISSUE TITLE
Cannot read property 'items' from undefined" despite all playlists and channels existing

### ISSUE THREAD

**Issue by Fabian42** [Elijas/auto-youtube-subscription-playlist-2#32]
Since yesterday, the script fails on every execution, with the error message "`TypeError: Cannot read property "items" from undefined. (line 169, file "Code")`". I checked all playlist, user and channel IDs. I have also replaced "ALL" with an explicit list of all user/channel names/IDs.

Any idea what it could be?

**Comment by y2kbugger** [comment]
Mine stopped working as well. there's also a weird side effect that on two channels of my subscriptions hundreds of videos were added dating back years. I haven't tried to debug deeply yet but I was suspecting something was going weird with their dates...

**Comment by Remco32** [comment]
> There's also a weird side effect that on two channels of my subscriptions hundreds of videos were added dating back years.

Same here. I used an auto clicker to 'manually' remove everything from the playlist and started over.

**Comment by deepfriedmind** [comment]
I had the same issue. I looked in the log and noticed `ERROR: API call to youtube.search.list failed with error: Daily Limit Exceeded.`. I now see that my dailes queries have risen astronomically and exceed the limit of 1,000,001. I have a trigger set to run the script every hour and haven't been close to reach the limit before.

**Comment by Fabian42** [comment]
I got some "execution time exceeded" errors yesterday, but then I added
some videos manually and set the timestamp in the sheet one day forward,
then it ran in under 5 minutes, but still took quite a while and gave out
the error message.

**Comment by deepfriedmind** [comment]
After the daily queries reset it's running again but now it's just going crazy adding hundreds of old videos and duplicates until the script times out.

**Comment by JesperDramsch** [comment]
> After the daily queries reset it's running again but now it's just going crazy adding hundreds of old videos and duplicates until the script times out. That explains the huge increase in daily quota.

Happens to me too.

**Comment by satishev** [comment]
+1, maybe it's not reading the date from the sheet correctly or calculating the difference correctly?

**Comment by Seiryuu-sama** [comment]
I also noticed the script getting errors, both time out and actual ones since a few day ago. I get different error however:
`TypeError: Second argument to Function.prototype.apply must be an array. (line 54, file "Code")`

I also noticed the timestamp is behind by 1 day

**Comment by Fabian42** [comment]
Try replacing "ALL" in your sheet with the individual subscriptions. For
some that solved the problem, for some it just caused a different error
message (the only this report is about).

**Comment by JesperDramsch** [comment]
https://support.google.com/youtube/thread/2494861?hl=en

Search is broken at the moment.

**Comment by ittaim** [comment]
Hi everyone,

TL;DR -
I've been using this tool happily for about a year but now it stopped working without any change on my side. I run "Update Playlists" but the destination playlist is not populated at all after the script is done.

I read the comments here and I'm trying to figure out if we experience the same issue -
I'm not running the automatic tool. I manually run the Functions-->Update Playlists feature (on almost a daily basis), with the "ALL" feature turned on. I read that someone suggested to change the "ALL" to something more specific - that also didn't work.

Does anyone know of any workaround for this issue? I highly depend on this tool and this is a real crisis lol

Thank you!

**Comment by Elijas** [comment]
Hi everyone! I'm really sorry to hear that the app is most likely broken at the moment, however I'm currently really busy with my other commitments, although I hope to find some spare time during the upcoming weekends, but count on me for that.

However, one of you created a bounty in Bountify, which will end in 7 days, to fix the recent major bug
https://bountify.co/auto-add-youtube-videos-from-subscribed-channels-to-specific-playlists

If you're a programmer, you can try to tackle it, if you're not, you could reach out to the author to raise the reward, or just spread the word!

**Comment by Elijas** [comment]
@JesperDramsch I've merged the PR, thank you!

It seems that youtube might have purposefully broken their search (due to the New Zealand event), which might have affected API search that we use too, if that's the case it's possible that all we can do is wait. Although, I didn't confirm this

**Comment by dudemario** [comment]
Youtube's search being down did affect this, and now that it's back up, everything should work fine.

**Comment by dudemario** [comment]
@Elijas can we close this due to being outdated?

**Comment by Elijas** [comment]
@dudemario Thanks!
